### ISSUE TITLE
Professor Layton: daily puzzles fail miserably

### ISSUE THREAD

**Issue by Molive-0** [citra-emu/citra#2160]
Hello!

This is something that needs to be addressed at somepoint so here it is:

Professor Layton has a thing called Daily Puzzles, in which your 3ds connects to the internet and downloads (you guessed it) daily puzzles for you to solve.
However, using these in citra renders some interesting results.
(FYI this was tested on PL6. I'm assuming this happens on PL5 also.)

The game runs normally (albeit laggily) up until the download screen, where you are presented with a shiny 'Download' button. Upon hitting this, it does none of the such.
The game waits a short amount of time, then a load of rubbish is spat out in the console and it softlocks.
If you wait a short while it'll hardlock for you also.

I've looked at the console and the most striking element was
```
Fatal error!
PID: 0x00000000_0x00000000
REV: 49580
AID: 0x00000000_0x00000000
ADR: 0x0014B014
RSL: 0xD8E007F7
Level: 27
Summary: 7
Module:1
Desc: 1015
```
The console also claims that there was several unknown / unimplemented commands.
At the end of the log the audio engine crashes alongside it too.

I didn't expect this to work, but I didn't expect the amount of junk thrown at me when it happened. Citra really didn't like it.

Anyway, thanks.
Molive

EDIT: happens in HEAD-fcd9a32
![welp](https://cloud.githubusercontent.com/assets/23285288/20399773/25217290-acea-11e6-9717-a92293d8e95c.png)

**Comment by JayFoxRox** [comment]
Please add a full log to the first post + add version information [citra commit of used version - should be current master]

(also pick a more descriptive issue title next tim [don't rename this])

**Comment by Molive-0** [comment]
Fixed

**Comment by Molive-0** [comment]
No, I meant I fixed the description

**Comment by AdmiralCurtiss** [comment]
The same thing happens for the bonus content in Professor Layton VS Phoenix Wright. If you switch the Audio backend to LLE it at least doesn't crash but the extra data downloading fails anyway. If you dump the extra data from a real system (I used Checkpoint) you instead get this error message:

![extra_data_corrupted](https://user-images.githubusercontent.com/4522237/62007816-63203300-b152-11e9-9638-39ad55730719.png)

Logfile for trying to load dumped extra data: https://gist.github.com/AdmiralCurtiss/35bbbb1006c94b8ad8eb5fedd8887d57
(relevant part around `[  25.870797]`; I'm guessing it's the stubbed BOSS stuff that makes it fail?)

**Comment by Dragios** [comment]
@AdmiralCurtiss For the extra data part it might been fixed by #5213. Could you test and let us know the result? Download the latest nightly should suffice.

**Comment by AdmiralCurtiss** [comment]
Nope, still acts the same to me. HLE Audio crashes, LLE audio gives the 'extra data is corrupted' message. Log looks about the same too.

I guess I'll try redumping the extra data to make sure it isn't actually corrupted but it doesn't look like that PR helped here.

**Comment by Dragios** [comment]
Is your Checkpoint version updated to latest by any chance? If it is this message can be safely discarded.

**Comment by AdmiralCurtiss** [comment]
Alright, so I went ahead and redumped the extdata from my 3DS using Checkpoint 3.7.4, and also made sure that the extdata works correctly as-is on actual hardware. (it does)

Using nightly-1536 on Windows with LLE (accurate) audio.

Without extra data on the emulated SD card, entering the Extra Content menu works fine, but then hitting 'Download Content' fails the download with an 'cannot access extra data' error ([a](https://user-images.githubusercontent.com/4522237/81440342-30d74380-9170-11ea-8efe-1bc712c0a331.png), [b](https://user-images.githubusercontent.com/4522237/81440425-55332000-9170-11ea-8601-19c02ce00850.png)) while it creates a few extradata metadata files on the emulated SD card. Retrying after that gives a different error message ([c](https://user-images.githubusercontent.com/4522237/81440476-6b40e080-9170-11ea-842a-bbdd13d29ac1.png); no system message) but no success either. That part is fine, I'm not expecting this to work anyway, but for completion's sake, maybe it gives some insight on what is wrong.

Rebooting using the created metadata and copying the dumped vs_sd1.bin into the created empty 'user' subfolder, I get 'The extra data is corrupted.' when the game attempts to load it from the SD card upon entering the Extra Content menu. ([d](https://user-images.githubusercontent.com/4522237/81440478-6bd97700-9170-11ea-8937-9fb5e5ee1f76.png), [e](https://user-images.githubusercontent.com/4522237/81440480-6bd97700-9170-11ea-9902-8627c101c268.png)) And, of course, none of the bonus content is unlocked after that error message.
